SharedExpensesApp/app/src/main/java/com/example/sharedexpensesapp/ui/screens/BalanceScreen.kt

```diff
@@ -1,77 +1,43 @@
 package com.example.sharedexpensesapp.ui.screens
 
-import android.util.Log
 import androidx.compose.foundation.Image
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.Row
-import androidx.compose.foundation.layout.Spacer
-import androidx.compose.foundation.layout.fillMaxWidth
-import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.*
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
 import androidx.compose.material3.CardDefaults
 import androidx.compose.material3.ElevatedCard
 import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.LaunchedEffect
-import androidx.compose.runtime.derivedStateOf
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
-import androidx.compose.runtime.setValue
+import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.colorResource
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
 import com.example.sharedexpensesapp.R
-import com.example.sharedexpensesapp.datasource.DebtsCallback
-import com.example.sharedexpensesapp.datasource.RestClient
-import com.example.sharedexpensesapp.datasource.UsersCallback
 import com.example.sharedexpensesapp.model.Debt
 import com.example.sharedexpensesapp.model.GroupItem
-import com.example.sharedexpensesapp.model.User
 
 @Composable
 fun BalanceScreen(
     modifier: Modifier = Modifier,
-    group: GroupItem?
+    group: GroupItem?,
+    balanceViewModel: BalanceViewModel = viewModel()
 ) {
-    var receivedDebts by remember { mutableStateOf(emptyList<Debt>()) }
-    var receivedUsers by remember { mutableStateOf(emptyList<User>()) }
     val receivedUsersMap by remember {
         derivedStateOf {
-            receivedUsers.associate { it.id to it.name }
+            balanceViewModel.users.associate { it.id to it.name }
         }
     }
 
     LaunchedEffect(Unit) {
```

Comment:
Maybe instead of calling LaunchedEffect twice, we can just put two methods inside?
```kotlin
 LaunchedEffect(Unit) {
    balanceViewModel.fetchDebts(group!!)
    balanceViewModel.fetchUsers(group!!)
}
```